README.md

```diff
@@ -59,30 +65,113 @@ https://code.launchpad.net/~ubuntu-core-dev/shim/+git/shim/+ref/master
 -------------------------------------------------------------------------------
 What patches are being applied and why:
 -------------------------------------------------------------------------------
-  * d/p/Fix-OBJ_create-to-tolerate-a-NULL-sn-and-ln.patch: Fix NULL
-    pointer dereference when calling OBJ_create() that leads to an
-    exception error on arm64. (LP: #1811901)
-  * debian/patches/MokManager-avoid-unaligned.patch: Fix compilation with GCC9:
-    avoid -Werror=address-of-packed-member errors in MokManager.
-  * debian/patches/tpm-correctness-1.patch,
-    debian/patches/tpm-correctness-2.patch: fix issues in TPM calls to ensure
-    the measurements are consistent with what is entered in the TPM event log.
-  * debian/patches/tpm-correctness-3.patch: Don't log duplicate identical
-    TPM events.
-  * debian/patches/MokManager-hidpi-support.patch: Do a little bit more to
-    try to get a more usable screen resolution for MokManager when running on
-    HiDPI screens; by trying to detect such cases and switching to mode 0.
-  * d/patches/fix-path-checks.patch: Cherry-pick upstream fix for regression
-    in loading fwupd, or anything else specified as an argument (LP: #1864223)
+
+  * debian/patches/359.patch: Make fallback boot noninteractive
+    without any delays. Currently a 5s delay is present on every boot
+    in the cloud with TPM, awaiting user to potentially interrupt and
+    directly boot instance with incorrect TPM measurements. In Ubuntu,
+    we can only realistically support reset after boot entries are
+    created to ensure firstboot userspace has correct TPM
+    measurements. Submitted as https://github.com/rhboot/shim/pull/359
+
+-------------------------------------------------------------------------------
+If bootloader, shim loading is, GRUB2: is CVE-2020-14372, CVE-2020-25632,
+ CVE-2020-25647, CVE-2020-27749, CVE-2020-27779, CVE-2021-20225, CVE-2021-20233,
+ CVE-2020-10713, CVE-2020-14308, CVE-2020-14309, CVE-2020-14310, CVE-2020-14311,
+ CVE-2020-15705, and if you are shipping the shim_lock module CVE-2021-3418
+-------------------------------------------------------------------------------
+grub2 2.04-1ubuntu45
+
+Patch set for EFI is from https://github.com/rhboot/grub2/commits/grub-2.02-sb
+Patches are all available in the repo for grub2:
+https://git.launchpad.net/~ubuntu-core-dev/grub/+git/ubuntu/tree/debian/patches?h=ubuntu
+... and marked "linuxefi_*" for the SecureBoot specific patches.
+
+Patches from Debian are not specially marked; but as the code is based on Debian's GRUB,
+Ubuntu-specific patches are marked "ubuntu-*"
+
+Also note the many CVE patches in the same directory.
+
+-------------------------------------------------------------------------------
+What exact implementation of Secureboot in GRUB2 ( if this is your bootloader ) you have ?
```

Comment:
s/you have/do you have/

Reply:
this bit needs to go into upstream repo, as that's from them.